rust/lance/src/index/vector/ivf.rs

```diff
@@ -407,13 +410,20 @@ impl Index for IVFIndex {
 impl VectorIndex for IVFIndex {
     #[instrument(level = "debug", skip_all, name = "IVFIndex::search")]
     async fn search(&self, query: &Query, pre_filter: Arc<PreFilter>) -> Result<RecordBatch> {
-        let partition_ids =
-            self.ivf
-                .find_partitions(&query.key, query.nprobes, self.metric_type)?;
+        let mut query = query.clone();
```

Comment:
how cheap is this copy, can we avoid it for the L2 path?

Reply:
This is very cheap, the query vector is zero copy (copied by pointer), other than that, there are just 6 other string/int fields.